### ISSUE TITLE
Python/Lua script "obs_data_get_*" returns unexpected results

### ISSUE THREAD

**Issue by coder39248583** [obsproject/obs-studio#4618]
### Operating System Info

Other

### Other OS

Windows 7 64bit

### OBS Studio Version

27.0.0-rc2

### OBS Studio Log URL

https://obsproject.com/logs/8gPiRIWX14HkbkMm

### Expected Behavior

A simple Python script should print the current state of a boolean property whenever its checkbox is toggled, as such:
False
True
False
True
...

### Current Behavior

The Python script outputs thus:
False
True
True
True
...

### Steps to Reproduce

**1)** Load the following python script:

```python
import obspython as obs

def script_properties():
    props = obs.obs_properties_create()
    obs.obs_properties_add_bool(props, "toggle", "Toggle")
    return props

def script_update(settings):
    print(obs.obs_data_get_bool(settings, "toggle"))
```

**2)** Repeatedly click the checkbox and view the log.

### Anything else we should know?

This also happens when adjusting an integer property, as with the following script:

```python
import obspython as obs

def script_properties():
    props = obs.obs_properties_create()
    obs.obs_properties_add_int(props, "integer", "Integer", 0, 100, 1)
    return props

def script_update(settings):
    print(obs.obs_data_get_int(settings, "integer"))
```

When incrementing the number 0, 1, 2, 3, etc., the log file prints the following:
0
1
1
1
...

I am running the recommended Python 3.6 64bit version (and OBS 64bit)

**Comment by ratwithacompiler** [comment]
Same issue with Lua on MacOS (x64) with 27-rc2.

```lua
obs = obslua

g_some_int = nil
g_enabled = nil

function script_properties()
    print("script_properties")
    local props = obs.obs_properties_create()
    obs.obs_properties_add_bool(props, "enabled", "Enabled")
    obs.obs_properties_add_int(props, "some_int", "Some int", 1, 100000, 1)
    return props
end

function script_defaults(settings)
    print("script_defaults")
    obs.obs_data_set_default_bool(settings, "enabled", true)
    obs.obs_data_set_default_int(settings, "some_int", 1000)
end

function script_update(settings)
    print("script_update")
    g_some_int = obs.obs_data_get_int(settings, "some_int")
    g_enabled = obs.obs_data_get_bool(settings, "enabled")

    print(("script_update: some_int: %d. enabled: %s"):format(g_some_int, g_enabled))
end

function script_load(settings)
    g_some_int = obs.obs_data_get_int(settings, "some_int")
    g_enabled = obs.obs_data_get_bool(settings, "enabled")
    print(("script_load some_int: %d. enabled: %s"):format(g_some_int, g_enabled))
end
```

works as expected on 26.1, only seems to register the first data update per prop on 27.

**Comment by WizardCM** [comment]
OK, this is caused by the UI itself. The bug was introduced in 86eb7aeb6935260725f91ff0b034995d4e7b4154 ("UI: Undo/Redo Properties and Filters").

![image](https://user-images.githubusercontent.com/941350/116804933-2a529080-ab66-11eb-9d34-1dbce156e799.png)

**Comment by jp9000** [comment]
Fixed in cec8e246303dd1db0adf7ca3d52604cedcdea182
